### PR TITLE
fix errors and move saver

### PR DESCRIPTION
**Pull request by Ahmkel**

### CODE REVIEW

File: models/example_model.py [hunk 1]
```diff
@@ -7,18 +7,16 @@
...
+        self.build_model()
 
-    def get_embedding_matrix():
+    def get_embedding_matrix(self):
         # Todo: implement by gammal
         return None
         
-    def embedding_layer():
+    def embedding_layer(self):
         embedding_matrix = get_embedding_matrix()
```
Comment (ghost) [inline models/example_model.py]: Shouldn't this be self.get_embedding_matrix() ?

File: models/example_model.py [hunk 2]
```diff
@@ -7,18 +7,16 @@
...
         
         return Embedding(
```
Comment (ghost) [inline models/example_model.py]: You have not imported the embedding layer from Keras.

File: models/example_model.py (hunk 1 above)
Comment (Ahmkel) [reply models/example_model.py]: yup

**Review by ghost — APPROVED** [review]

File: trainers/example_trainer.py [hunk 3]
```diff
@@ -6,8 +6,22 @@
 class ExampleTrainer(BaseTrain):
     def __init__(self, sess, model, data, config, logger):
         super(ExampleTrainer, self).__init__(sess, model, data, config,logger)
+        init_saver()
```
Comment (ghost) [inline trainers/example_trainer.py]: Should also be self.init_saver()

File: trainers/example_trainer.py [hunk 4]
```diff
@@ -6,8 +6,22 @@
...
+
+    def init_saver(self):
+        self.callbacks.append(
+            ModelCheckpoint(
```
Comment (ghost) [inline trainers/example_trainer.py]: Should import ModelCheckpoint from Keras

**Pull request merged**